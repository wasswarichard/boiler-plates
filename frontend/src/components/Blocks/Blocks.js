import React, {useEffect, useState} from "react";
import useFullPageLoader from "../../hooks/useFullPageLoader/useFullPageLoader";
import config from "../../Helpers/config.json";
import axios from "axios";
import Table from "../../utilities/DataTable/Table/Table";
import store from "../../store/store";
import {AddBlocks} from "../../actions/actions"
import PreviewModal from "../PreviewModal/PreviewModal";
import useSWR from 'swr';

const Blocks = () => {
    const [blocks, setBlocks] = useState([]);
    const [loader, showLoader, hideLoader] = useFullPageLoader();
    const [modalInfo, setModalInfo] = useState({})
    const [show, setShow] = useState(false)

    useEffect(() => {
       modalInfo.hash && setShow(true)
    }, [modalInfo])

    const handleClose = () => {
        setShow(false)
    }

    const headers = [
        {name: "Hash", field: "hash", type: "input"},
        {name: "Height", field: "height", type:"input"},
        {name: "Time", field: "time", type: "date"},
        {name: "Block Index", field: "block_index", type: 'input'}
    ]
    useEffect(() => {
        showLoader();
        const loadBlocks = async () => {
            await axios.get(`${config.apiUrl}/blocks/v1a/blocks`)
                .then(response => {
                    hideLoader();
                    store.dispatch(AddBlocks({...response.data}))
                    setBlocks(response.data);
                }).catch(error => {
                    hideLoader();
                })
        };
        loadBlocks();
    }, []);

    return (
        <div className="main-content">
            <main>
                <Table headers={headers} loader={loader} blocks={blocks} setModalInfo={setModalInfo}/>
            </main>
            {show ? <PreviewModal show={show} modelInfo={modalInfo} handleClose={handleClose}/> : null}
        </div>
    )
}
export default Blocks;