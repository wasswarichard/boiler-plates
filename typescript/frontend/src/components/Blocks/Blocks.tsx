import React from "react";

const Blocks = () => {
    return (
        <div className="main-content">
            Hello
        </div>
    )
}
export default Blocks;