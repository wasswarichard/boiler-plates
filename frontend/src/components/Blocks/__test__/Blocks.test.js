import React from "react";
import ReactDOM from "react-dom";
import Blocks from "../Blocks";
import { render, cleanup} from "@testing-library/react";
import TestRenderer from 'react-test-renderer';

afterEach(cleanup)
it('should render without crashing', function () {
    const div = document.createElement("div")
    ReactDOM.render(<Blocks></Blocks>, div)
});

it('should render the blocks correctly', function () {

});

it('should match snapshot', function () {
    const tree = TestRenderer.create(<Blocks></Blocks>).toJSON();
    expect(tree).toMatchSnapshot();

});
