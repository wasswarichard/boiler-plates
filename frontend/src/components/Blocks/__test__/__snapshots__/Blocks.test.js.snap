// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div
  className="main-content"
>
  <main>
    <div>
      <div
        className="row"
      >
        <div
          className="col-md-6"
        >
          <nav>
            <ul
              className="pagination"
            />
          </nav>
        </div>
      </div>
      <div
        style={
          Object {
            "overflow": "auto",
          }
        }
      >
        <table
          cellSpacing="0"
          className="table table-striped table-bordered table-sm"
          id="dtBasicExample"
          width="50%"
        >
          <thead>
            <tr>
              <th>
                Hash
              </th>
              <th>
                Height
              </th>
              <th>
                Time
              </th>
              <th>
                Block Index
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td
                className="th-sm"
              >
                <input
                  className="form-control"
                  placeholder="Hash"
                  rk-model="hash"
                  style={
                    Object {
                      "width": "200px",
                    }
                  }
                  type="input"
                />
              </td>
              <td
                className="th-sm"
              >
                <input
                  className="form-control"
                  placeholder="Height"
                  rk-model="height"
                  style={
                    Object {
                      "width": "200px",
                    }
                  }
                  type="input"
                />
              </td>
              <td
                className="th-sm"
              >
                <input
                  className="form-control"
                  placeholder="Time"
                  rk-model="time"
                  style={
                    Object {
                      "width": "200px",
                    }
                  }
                  type="date"
                />
              </td>
              <td
                className="th-sm"
              >
                <input
                  className="form-control"
                  placeholder="Block Index"
                  rk-model="block_index"
                  style={
                    Object {
                      "width": "200px",
                    }
                  }
                  type="input"
                />
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </main>
</div>
`;
